src/stactools/datacube/stac.py

```diff
@@ -94,8 +94,8 @@ def read_dimensions_and_variables(
 
         diff = np.diff(data)
         if len(diff) > 1:
-            evenly_spaced = cast(bool, np.all(diff == diff[0]))
-            step = float(data[1] - data[0]) if evenly_spaced else None
+            evenly_spaced = np.allclose(diff, np.mean(diff))
```

Comment:
Should probably allow configurable tolerance, especially if the spacing could be a variety of units.

Reply:
I generally prefer explicit arguments unless there's a good reason for something to be an envvar.

Reply:
Now there is a `--rtol` flag to trigger this behavior.